plugins/org.wso2.developerstudio.visualdatamapper.diagram/src/org/wso2/developerstudio/datamapper/diagram/edit/parts/DataMapperRootEditPart.java

```diff
@@ -16,65 +16,319 @@
+            String attr1 = mappings.get(i*4);
+            String path1 = mappings.get(i*4 + 1);
+            String attr2 = mappings.get(i*4 + 2);
+            String path2 = mappings.get(i*4 + 3);
+            
+            Iterator iterator = inputTreeNodesList.iterator();
+            String nodeName = "";
+            TreeNode treeNodeFin = null; 
+            TreeNode treeNode = iterateList(treeNodeFin, iterator, attr1, path1);
+            
+            
+            Iterator iteratorOut = outputTreeNodesList.iterator();
+            nodeName = "";
+            TreeNode treeNodeOutFin = null;
+            TreeNode treeNodeOut = iterateList(treeNodeOutFin, iteratorOut, attr2, path2);
+            
+            try {
+                final EObject source = (EObject) treeNode.getOutNode();
+                final EObject target = (EObject) treeNodeOut.getInNode();
+                final OutNode container = deduceContainer(source, target);
+            
+            
+                final DataMapperLink newElement = DataMapperFactory.eINSTANCE.createDataMapperLink();
+                TransactionalEditingDomain domain = TransactionUtil.getEditingDomain(container);
+                try {
+                    Thread.sleep(2);
+                } catch (InterruptedException e) {
+                    // TODO Auto-generated catch block
+                    e.printStackTrace();
+                }
+                domain.getCommandStack().execute(new RecordingCommand(domain) {
+                    @Override
+                    protected void doExecute() {
+                        container.getOutgoingLink().add(newElement);
+                        newElement.setOutNode((OutNode) source);
+                        newElement.setInNode((InNode) target);
+                    }
+                });
+            } catch (Exception e) {
+                continue;
+            }
+        }
+        
+        UpdateDiagramCommand updater = new UpdateDiagramCommand();
+        try {
+            updater.execute(null);
+        } catch (ExecutionException e) {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
+        }
+    }
+
+    private TreeNode iterateList(TreeNode treeNodeFinal, Iterator iterator, String attributeName, String attributePath) {
+        TreeNode treeNode = null;
+        String nodeName = null;
+        while(iterator.hasNext() && !attributeName.equals(nodeName)) {
+            treeNode = (TreeNode) iterator.next();
+            nodeName = (String) treeNode.getName();
+            if (attributeName.equals(nodeName)) {
+                treeNodeFinal = treeNode;
+            }
+            ////////////////////////
+            if ( attributePath.indexOf(nodeName) !=-1) {
+                treeNodeFinal = traverseNode(treeNodeFinal, treeNode, attributeName, attributePath);
+            }
+            ///////////////////////////
+//          treeNodeFinal = traverseNode(treeNodeFinal, treeNode, attributeName);
```

Comment:
Shall we remove the commented code segements?

Reply:
Done. Please check